src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -757,9 +768,14 @@ class HttpGoogleServicesDAO(
   }
 
   private def updateBindings(bindings: Seq[Binding], template: ProjectTemplate) = {
-    bindings.map { policy =>
-      val newMembers = (template.policies.getOrElse(policy.getRole, Seq.empty) ++ policy.getMembers).distinct
-      new Binding().setRole(policy.getRole).setMembers(newMembers)
+    val templateMembersWithRole = template.policies.toSeq
+    val existingMembersWithRole = bindings.map { policy => policy.getRole -> policy.getMembers.toSeq }
+    val updatedPolicies = (templateMembersWithRole ++ existingMembersWithRole).
+      groupBy { case (role, _) => role }.
+      map { case (role, membersWithRole) => role -> membersWithRole.map { case (_, members) => members }.flatten }
+
+    updatedPolicies.map { case (role, members) =>
+      new Binding().setRole(role).setMembers(members)
```

Comment:
the old code above only set roles that were already in bindings but what we really need is all the roles in bindings plus any additional roles in the template